File: src/main/java/com/medicalsystem/util/ExcelUtils.java
package com.medicalsystem.util;

import org.apache.poi.ss.usermodel.*;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import java.io.FileInputStream;
import java.io.IOException;

public final class ExcelUtils {

    private static final DataFormatter formatter = new DataFormatter();
    private static CellStyle dateCellStyle = null;

    private ExcelUtils() {}

    public static Workbook loadWorkbook(String excelFileName) throws IOException {
        return new XSSFWorkbook(new FileInputStream(excelFileName));
    }

    public static String getValueAsString(Cell cell) {
        return formatter.formatCellValue(cell);
    }

    public static void setDateCellStyle(Cell cell) {
        if (dateCellStyle == null) {
            Workbook workbook = cell.getRow().getSheet().getWorkbook();
            dateCellStyle = workbook.createCellStyle();
            CreationHelper creationHelper = workbook.getCreationHelper();
            dateCellStyle.setDataFormat(creationHelper.createDataFormat().getFormat("dd.mm.yyyy"));
        }
        cell.setCellStyle(dateCellStyle);
    }
}

File: src/main/kotlin/com/medicalsystem/excel/ExcelUtils.kt
package com.medicalsystem.excel

import org.apache.poi.ss.usermodel.Cell
import org.apache.poi.ss.usermodel.CellStyle
import org.apache.poi.ss.usermodel.DataFormatter
import org.apache.poi.ss.usermodel.Workbook

private val formatter = DataFormatter()
private var dateCellStyle: CellStyle? = null

/**
 * Returns cells content as string.
 */
fun Cell.getAsString(): String = formatter.formatCellValue(this).trim()

/**
 * Makes the cell display dates properly.
 */
fun Cell.setStyleAsDate() {
    if (dateCellStyle == null) {
        val workbook: Workbook = this.row.sheet.workbook
        dateCellStyle = workbook.createCellStyle()
        dateCellStyle!!.dataFormat = workbook.creationHelper.createDataFormat().getFormat("dd.mm.yyyy")
    }
    this.cellStyle = dateCellStyle!!
}